include/drivers/sdhc.h

```diff
@@ -0,0 +1,437 @@
+/*
+ * Copyright (c) 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief SD Host Controller public API header file.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_SDHC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_SDHC_H_
+#include <device.h>
+#include <sd/sd_spec.h>
+
+/**
+ * @brief SDHC interface
+ * @defgroup sdhc_interface SDHC interface
+ * @ingroup io_interfaces
+ */
+
+#ifdef _cplusplus
+extern "C" {
+#endif
+
+
+/**
+ * @name SD command timeouts
+ * @{
+ */
+#define SDHC_TIMEOUT_FOREVER (-1)
+/** @} */
+
+/**
+ * @brief SD host controller command structure
+ *
+ * This command structure is used to send command requests to an SD
+ * host controller, which will be sent to SD devices.
+ */
+struct sdhc_command {
+	uint32_t opcode; /*!< SD Host specification CMD index */
+	uint32_t arg; /*!< SD host specification argument */
+	uint32_t response[4]; /*!< SD card response field */
+	enum sdhc_rsp_type response_type; /*!< Expected SD response type */
+	unsigned int retries; /*!< Max number of retries */
+	int timeout_ms; /*!< Command timeout in milliseconds */
```

Comment:
I would like to allow the user to pass a special "forever" value to the timeout, like `K_FOREVER` for kernel APIs. Would you suggest moving to `k_timeout_t` for this field in that case?

Reply:
I think it is implementation dependent, probably yes.

Reply:
I guess it would simplify things like

```c
	if (cmd->timeout_ms == SDHC_TIMEOUT_FOREVER) {
		request.command_timeout = K_FOREVER;
	} else {
		request.command_timeout = K_MSEC(cmd->timeout_ms);
	}
```

Reply:
Do you mean by adding a layer like this to the `sdhc.h` file, that checks the timeout set on the command and adds it to the request? The other approach I'm considering is having a default timeout, since it is rarely desirable to have a zero timeout command (usually the user wants a command with a short, but nonzero timeout). The user could then override the timeout only when desired.